spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/GqlDatastoreQuery.java

```diff
@@ -267,10 +288,84 @@ boolean isNonEntityReturnedType(Class returnedType) {
 			this.finalGql = spelEvaluator.getQueryString();
 
 			for (Map.Entry<String, Object> entry : results.entrySet()) {
-				this.params.add(entry.getValue());
-				// Cloud Datastore requires the tag name without the
+				Object value = entry.getValue();
+				this.params.add(value);
+				// Cloud Datastore requires the tag name without the @
 				this.tagsOrdered.add(entry.getKey().substring(1));
 			}
+
+			ParameterAccessor paramAccessor =
+					new ParametersParameterAccessor(getQueryMethod().getParameters(), rawParams);
+			Sort sort = paramAccessor.getSort();
+			if (!sort.equals(Sort.unsorted())) {
+				this.finalGql = addSort(this.finalGql, sort);
+			}
+			this.noLimitQuery = this.finalGql;
+			Pageable pageable = paramAccessor.getPageable();
+			if (!pageable.equals(Pageable.unpaged())) {
+				this.finalGql += LIMIT_CLAUSE;
```

Comment:
Consider StringBuilder append

Reply:
it get's optimized to StringBuilder by compiler anyway